artifact-manifest/artifact-manifest.md

```diff
@@ -0,0 +1,462 @@
+
+The following command would copy the `mysql:8` image from docker hub to the acme-rockets registry. The CLI could be run within the source or target cloud. 
+
+```bash
+oci-reg copy \
+  --source hub.docker.io/mysql:8 \
+  --target registry.acme-rockets.io/base-artifacts/mysql:8
+```
+
+The `oci-reg copy` command would:
+
+- assure the manifest and layer/blob digests remain the same
+- copy any artifacts that are dependent on the source artifact-manifest, persisting them in the target registry.
+
+### Reference Artifacts
+
+There are a set of artifact types that declare references to other artifacts that may, or may not be stored in the same registry. The reference is important to note, indicating copying to be capable between environments, as well as generalized validation scenarios.
+
+#### Helm Reference
+
+![mysql image copy](./media/wordpress-helm-chart-copy.svg)
+
+In the above scenario, a helm chart is copied from a public registry to the ACME Rockets registry.  The `wordpress-chart:v5` is represented as an `application/vnd.oci.artifact.manifest.v1+json`. The `wordpress-chart:v5` helm chart references the `wordpress:v5` image and the `mysql:8` image. All three artifacts have signatures attesting to their authenticity.
+
+As the copy is initiated, the `oci.artifact.manifest` of the `wordpress-chart:v5` is evaluated. As the chart references the same version (digest) of the `mysql:8` image already in theACME Rockets registry, the copy skips duplicating the content and moves to copying the `wordpress:v5` image, the `wordpress-chart:v5` and their associated signatures.
+
+In this case, the images referenced in the chart existed in the source registry. However, this may not be the case in all scenarios. A helm chart may be acquired from a different location, referencing images from docker hub. This deferred validation is left to client tools, enabling common package management resolution scenarios.
+
+To support the loose references between artifacts, a `references` collection is added to the `oci.artifact.manifest`:
+
+**A `wordpress-chart:v5` Helm Chart example:**
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.artifact.manifest.v1+json",
+  "artifactType": "application/vnd.cncf.helm.v3",
+  "config": {
+    "mediaType": "application/vnd.cncf.helm.config.v1+json",
+    "size": 0,
+    "digest": "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7"
+  },
+  "blobs": [
+    {
+      "mediaType": "application/vnd.cncf.helm.chart.v1.tar",
+      "digest": "sha256:9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f0",
+      "size": 32654
+    },
+    {
+      "mediaType": "application/vnd.cncf.helm.values.v1.yaml",
+      "digest": "sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b",
+      "size": 16724
+    }
+  ],
+  "references": [
+    {
+      "artifact": "wordpress:5.7",
```

Comment:
I don't like tags in here. Not sure about repositories...

Does the repo refer to a sibling repository? Child repository? Top-level?

Can this just be under a `org.opencontainers.image.ref.name` annotation?

Reply:
I'll have an update later this week that will use a mix of annotations and descriptors to cleanup the references.

Reply:
The more I think about the repository thing, the more I dislike it. Repositories are a natural security boundary. You're introducing a multi-repository artifact, which presents a lot of problems in terms of authentication and management.

We removed catalog because it violated these boundaries.

Cross-repository mounting is the only thing similar to this that remains, and the authentication for that is really hard to get right. This worries me.

Reply:
So, a helm chart, a CNAB and other artifacts are already cross repository artifact types. But, we are describing these as loose references that the client can validate independently. The `blobs` and manifests that have a `depends-on` annotation are hard dependencies and must be in the same repo, and must exist to complete the manifest put. 
So, I think we're straddling this problem, supporting the artifact types, while not putting an undue burden on auth boundaries that don't already exist today.

Reply:
What would the client do with a helm cross repository artifact reference?

I ask since the values for the helm template could override the default tag or even use a different registry server, and we can't (or at least shouldn't) template the loose reference in the registry, so I feel like any helm client should ignore this field. If we use the field for signing, does a different helm values invalidate the signed chart? If we use the field for mirroring, would mirroring tools potentially copy images that go unused?

I'm having a hard time seeing the value add for including a potentially templated field into a non-templated registry object that doesn't introduce the risk of breakage.